### DISCUSSION TITLE
Email Extended Scalar Type

### DISCUSSION THREAD

**Discussion by willisjtc** [graphql-java/graphql-java#2812] [Ideas]
Have there been any discussions on whether or not an Email scalar type would be supported?

**Comment by bbakerman** [comment]
We would not put a email scalar into graphql-java the base engine.  

However the librray https://github.com/graphql-java/graphql-java-extended-scalars might be a candidate.

That said email is one of the most notoriously hard  "scalars" to get right - to parse an email in and say "its acceptable" is something that 1000s of projects have tried and I am personally havent seen great success in.

It maybe that there is a great algo / lib for email validation - but I havent seen it - but if there is - then it could be a scalar in extended scalars in theory

**Reply by willisjtc** [reply]
hey @bbakerman , I was thinking graphql-java-extended-scalars as well. I was about to put an issue in that repo but it said for questions it might be better to bring something up in discussions in this repo.

I agree it's difficult thing to get right, so my proposal was to use the apache commons email validator: https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/EmailValidator.html

though I'm not sure what the policies are on 3rd party dependencies

**Reply by bbakerman** [reply]
Ideally we would no extra dependency for one scalar.  So since this is Apache code, it could be copied and attributed inside this repo.

Its really not a big class

https://github.com/apache/commons-validator/blob/master/src/main/java/org/apache/commons/validator/EmailValidator.java
